pip_faster.py

```diff
@@ -383,12 +383,8 @@ def run(self, options, args):
                 options, args,
             )
 
-        if requirement_set is None:
-            required = ()
-            successfully_installed = ()
-        else:
-            required = requirement_set.requirements.values()
-            successfully_installed = requirement_set.successfully_installed
+        required = requirement_set.requirements.values()
+        successfully_installed = requirement_set.successfully_downloaded
```

Comment:
I'm not 100% confident in this change, need to look into it more.

Two things going on:

* `successfully_installed` isn't on `RequirementSet` anymore, and the list of installed things doesn't seem to be made available at all (not returned in any way or attached to an object we have): https://github.com/pypa/pip/blob/60208552084ec16bf70b1d2913aed61cccdbe28d/src/pip/_internal/commands/install.py#L327-L337; successfully_downloaded makes the tests pass, but I didn't confirm that there are no edge cases.
* `requirement_set` is never `None` anymore? Need to confirm that, but the tests were never hitting the first branch.

Reply:
I couldn't find any examples where `extraneous` was calculated differently based on subtracting out `requirement_set.successfully_installed` even on old venv-update (pre-pip-10); all the tests pass even after I calculate `extraneous` both with and without that subtraction and then assert their equality.

I've deleted that from the code for now since I can't find any reason not to.

@bukzor do you know by any chance if/why it's necessary?